### PR TITLE
Migrate linter definitions 3

### PR DESCRIPTION
**Pull request by TylerJang27**
PR is dependent on #78  

Add linter definitions for:
- actionlint
- ansible-lint
- brakeman
- clang-tidy
- clippy
- eslint
- gitleaks
- golangci-lint
- hadolint
- tflint
- terraform (lint + validate)
- semgrep

Also provides the following fixes/changes:
- Better onboarding: When no snapshots exist, rather than running against latest, run against KnownGoodVersion. This will provide a better experience since CI expects a KnownGoodVersion snapshot to exist.
- Adds a file-triggered action to generate boilerplate code with plugin.yaml and tests whenever someone creates a new linter subdirectory.
- Fixes naming convention when normalizing snapshot names to use underscores rather than dashes for linter names.
- Fixes semver regex matching for versionless linters
- Fixes semver regex matching for versions >=10 (was previously greedily sorted)

### CODE REVIEW

File: linters/ansible-lint/test_data/jboss-standalone/LICENSE.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Copyright (C) 2013 AnsibleWorks, Inc.
+
+The work in this jboss-standalone subdirectory is licensed under the Creative Commons Attribution 3.0 Unported License. To view a copy of
```
Comment (TylerJang27) [inline linters/ansible-lint/test_data/jboss-standalone/LICENSE.md:3]: Please advise on the content of this test subdirectory

File: linters/ansible-lint/readme.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# Ansible-Lint
+
+[Ansible-lint](https://github.com/ansible/ansible-lint) is used to check ansible playbooks. In order
```
Comment (TylerJang27) [inline linters/ansible-lint/readme.md:3]: Please advise if these trigger guidelines are sufficient

File: linters/ansible-lint/plugin.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+          success_codes: [0, 2]
+          run_linter_from: directory
+        - name: lint
+          version: ">=5.1.3"
+          # Custom parser type defined in the trunk cli to handle ansible-lint's output.
+          output: ansible_lint
+          # parseable-severity was removed after 5.4.0
+          run: ansible-lint --parseable-severity
+          success_codes: [0, 2]
+          run_linter_from: directory
```
Comment (TylerJang27) [inline linters/ansible-lint/plugin.yaml:22]: Note these changes/fixes

File: linters/rubocop/plugin.yaml [hunk 4]
```diff
@@ -25,20 +25,3 @@ lint:
...
-      package: rubocop
-      success_codes: [0, 1]
-      formatter: true
-      known_good_version: 1.39.0
-      version_command:
-        parse_regex: ${semver}
-        run: rubocop --version
-      in_place: true
-      batch: true
-      deprecated: rubocop-fmt is now handled by rubocop. Please delete rubocop-fmt from your config
```
Comment (TylerJang27) [inline linters/rubocop/plugin.yaml:44]: Because this is deprecated, I'm removing it from the plugin repo

File: linters/semgrep/plugin.yaml [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+        - name: check
+          output: sarif
+          # We use include filters instead of explicit paths because otherwise semgrep will run
+          # linters for every possible language for files that it does not recognize an extension for.
+          run: semgrep --config=auto --sarif --output=${tmpfile} --include=${target}
+          read_output_from: tmp_file
+          success_codes: [0]
+          batch: true
+          cache_results: true
+          # TODO(Chris): We need a new configuration option which means copy files into a sandbox.
```
Comment (TylerJang27) [inline linters/semgrep/plugin.yaml:18]: There's a couple todos like this one that remain. Should we make tickets to make sure they're tracked?

File: linters/sqlfluff/plugin.yaml [hunk 6]
```diff
@@ -23,19 +23,3 @@ lint:
           in_place: true
           success_codes: [0]
           enabled: false
-
-    - name: sqlfluff-fix
-      files: [sql, sql-j2, dml, ddl]
-      runtime: python
-      package: sqlfluff
-      deprecated:
```
Comment (TylerJang27) [inline linters/sqlfluff/plugin.yaml:31]: Because this is deprecated, I'm removing it from the plugin repo

File: linters/stylelint/plugin.yaml [hunk 7]
```diff
@@ -31,31 +31,3 @@ lint:
       version_command:
         parse_regex: ${semver}
         run: stylelint --version
-
-    - name: stylelint-fmt
-      files: [css, sass]
-      type: rewrite
-      deprecated:
```
Comment (TylerJang27) [inline linters/stylelint/plugin.yaml:38]: Because this is deprecated, I'm removing it from the plugin repo

File: linters/ansible-lint/readme.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+  enabled:
+    - ansible-lint@5.3.2
+
+  triggers:
+    # Run these linters
+    - linters:
+        - ansible-lint
+      # If any files matching these change
+      paths:
+        - "**"
```
Comment (det) [inline linters/ansible-lint/readme.md:19]: This currently says to run ansible lint on that subdir when any file changes, which may not make sense. Suggest the following change.

```suggestion
        - "test_data/jboss-standalone/**"
```

File: linters/semgrep/plugin.yaml (hunk 5 above)
Comment (det) [reply linters/semgrep/plugin.yaml:18]: I'll make an issue, thanks.

**Review by det — APPROVED** [review]
lgtm

**Pull request merged**